### PR TITLE
Feature cms search

### PR DESCRIPTION
**Pull request by kimakan**

### CODE REVIEW

File: daiquiri/files/views.py [hunk 1]
```diff
@@ -36,3 +45,27 @@ def get(self, request, file_path, **kwargs):
...
+
+    root = None
+
+    def get(self, request, **kwargs):
+
+        search_string = request.GET.get("q", "")
+
+        results = Searcher.search_for_string(string_query=search_string)
+
+        paginator = Paginator(results, 5)
```
Comment (agy-why) [inline daiquiri/files/views.py:60]: make the number of results in the search page to be in SETTINGS.

**Review by agy-why — COMMENTED** [review]
view.py line 29 :

```python
            file_path = self.root.strip('/') + '/' + file_path
```
            
use join please.

File: daiquiri/files/search.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+from django.utils.html import strip_tags
+from django.utils.text import Truncator
+
+from .utils import get_file_path, read_file_content
+
+logger = logging.getLogger(__name__)
+
+
+class Searcher:
+
```
[the remark this answers is not in the document]
Comment (kimakan) [reply daiquiri/files/search.py:19]: Done!

File: daiquiri/files/search.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+                title = html[title_starts_at:title_ends_at]
+        return title
+
+
+
+    @classmethod
+    def build_lunr_index(cls):
+
+        if settings.FILES_DOCS_REL_PATH is None:
+            logger.debug('FILES_DOCS_REL_PATH is not defined')
```
[the remark this answers is not in the document]
Comment (kimakan) [reply daiquiri/files/search.py]: Changed to `logger.error` and added `raise Http404`. Now, if the `FILES_DOCS_REL_PATH` is not defined or wrong, the user get 404.

File: daiquiri/files/views.py (hunk 1 above)
Comment (kimakan) [reply daiquiri/files/views.py]: Done!

File: daiquiri/files/search.py [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+            match_end = re.search(end_pattern, html[title_starts_at:])
+            if match_end:
+                title_ends_at = title_starts_at + match_end.span()[0]
+                title = html[title_starts_at:title_ends_at]
+        return title
+
+
+
+    @classmethod
+    def build_lunr_index(cls):
```
[the remark this answers is not in the document]
Comment (kimakan) [reply daiquiri/files/search.py:47]: Changed

**Comment by kimakan** [comment]
I've applied all requested changes/comments.

**Review by agy-why — COMMENTED** [review]
fine

**Pull request merged**